tests/test_mongodb_fname_restrictions.py

```diff
@@ -0,0 +1,285 @@
+
+            object_id = client['simple_db']['simple_coll_2'].find()[last_index]['_id'] # int.field 99
+            changed_ids.add(object_id)
+            client["simple_db"]["simple_coll_2"].update_one({'_id': object_id},{'$set': {'int.field': -1}})
+
+            # Insert two documents for each collection
+            last_index = client['simple_db']['simple_coll_1'].find().count()
+            client["simple_db"]["simple_coll_1"].insert_one({"int.field": 50, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_1"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index += 1 # inserting a new item
+            client["simple_db"]["simple_coll_1"].insert_one({"int.field": 51, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_1"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index = client['simple_db']['simple_coll_2'].find().count()
+            client["simple_db"]["simple_coll_2"].insert_one({"int.field": 100, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_2"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+            last_index += 1
+            client["simple_db"]["simple_coll_2"].insert_one({"int.field": 101, "string$field": random_string_generator()})
+            object_id = client["simple_db"]["simple_coll_2"].find()[last_index]['_id']
+            changed_ids.add(object_id)
+
+        #  -------------------------------------------
+        #  ----------- Subsequent Oplog Sync ---------
+        #  -------------------------------------------
+
+        # Run sync
+
+        sync_job_name = runner.run_sync_mode(self, conn_id)
+
+        exit_status = menagerie.get_exit_status(conn_id, sync_job_name)
+        menagerie.verify_sync_exit_status(self, exit_status, sync_job_name)
+
+
+        # verify the persisted schema was correct
+        messages_by_stream = runner.get_records_from_target_output()
+        records_by_stream = {}
+        for stream_name in self.expected_sync_streams():
+            records_by_stream[stream_name] = [x for x in messages_by_stream[stream_name]['messages'] if x.get('action') == 'upsert']
+
+
+        # assert that each of the streams that we synced are the ones that we expect to see
+        record_count_by_stream = runner.examine_target_output_file(self,
+                                                                   conn_id,
+                                                                   self.expected_sync_streams(),
+                                                                   self.expected_pks())
+
+        # Verify that we got at least 6 records due to changes
+        # (could be more due to overlap in gte oplog clause)
+        for k,v in record_count_by_stream.items():
+            self.assertGreaterEqual(v, 6)
+
```

Comment:
I think there should be another assertion to validate the document values in the collections so we ensure the special characters are read parsed correctly by the tap.

Reply:
Will add